### PR TITLE
python docs for Datatype-related classes

### PR DESCRIPTION
**Pull request by yoni206**

### CODE REVIEW

File: docs/api/python/python.rst [hunk 1]
```diff
@@ -12,5 +12,9 @@ Python API Documentation
 .. toctree::
     :maxdepth: 1
 
-    datatype
     quickstart
+    datatype
+    datatypeconstructor
+    datatypeconstructordecl
+    datatypedecl
+    datatypeselector
```
Comment (nafur) [inline docs/api/python/python.rst:20]: Did you forget to include these files?

**Review by nafur — COMMENTED** [review]
Again the indentation is weird. You may want to check automatic formatting (e.g. with `yapf`), I'd guess that it would take care of this.

File: docs/api/python/python.rst (hunk 1 above)
Comment (yoni206) [reply docs/api/python/python.rst:20]: Yes, now fixed :)

**Comment by yoni206** [comment]
Thanks for looking at this. I have a problem running `yapf` on this (maybe due to cython?). But I fixed the indentations manually, so now your comments are addressed.

File: src/api/python/cvc5.pxi [hunk 2]
```diff
@@ -207,32 +208,57 @@ cdef class DatatypeConstructor:
...
+        """
+            :return: the constructor operator as a term.
+        """
         cdef Term term = Term(self.solver)
         term.cterm = self.cdc.getConstructorTerm()
         return term
 
     def getSpecializedConstructorTerm(self, Sort retSort):
+        """
+            Specialized method for parametric datatypes (see :cpp:func:`DatatypeConstructor::getSpecializedConstructorTerm() <cvc5::api::Datatype::getSpecializedConstructorTerm>`).
```
Comment (nafur) [inline src/api/python/cvc5.pxi:226]: This link does not work, apparently. Also, it says `DatatypeConstructor` but links to `Datatype`. Should it go to `getConstructorTerm`?

File: src/api/python/cvc5.pxi [hunk 3]
```diff
@@ -251,16 +277,26 @@ cdef class DatatypeConstructor:
...
     cdef Solver solver
 
     def __cinit__(self, Solver solver):
         self.solver = solver
 
     def addSelector(self, str name, Sort sort):
+        """
+            Add datatype selector declaration.
+            :param name: the name of the datatype selector declaration to add.
+            :param sort: the range sort of the datatype selector declaration to add.
```
Comment (nafur) [inline src/api/python/cvc5.pxi:291]: Not sure what is wrong here, but the parameters are not properly rendered. (also in other places).

File: src/api/python/cvc5.pxi [hunk 4]
```diff
@@ -271,21 +307,35 @@ cdef class DatatypeConstructorDecl:
...
 
     def isParametric(self):
+        """
+            :return: is this datatype declaration parametric?
+        """
         return self.cdd.isParametric()
 
     def getName(self):
+        """
+            :return the name of this datatype declaration.
```
Comment (nafur) [inline src/api/python/cvc5.pxi:337]:
```suggestion
            :return: the name of this datatype declaration.
```

File: src/api/python/cvc5.pxi [hunk 5]
```diff
@@ -207,32 +208,57 @@ cdef class DatatypeConstructor:
...
+        """
         cdef Term term = Term(self.solver)
         term.cterm = self.cdc.getConstructorTerm()
         return term
 
     def getSpecializedConstructorTerm(self, Sort retSort):
+        """
+            Specialized method for parametric datatypes (see :cpp:func:`DatatypeConstructor::getSpecializedConstructorTerm() <cvc5::api::Datatype::getSpecializedConstructorTerm>`).
+            :param retSort: the desired return sort of the constructor
+                :return: the constructor operator as a term.
```
Comment (yoni206) [inline src/api/python/cvc5.pxi:228]: indentation, and doesn't render well.

**Pull request merged**